docs/related/async-raml-datatypes.mdx

```diff
@@ -0,0 +1,324 @@
+
+:::caution When referencing to external yaml/json files
+the external file **must not** be a known RAML fragment or api specification.
+:::
+
+The following case is a **valid** reference making use of relative json pointers.
+
+<div className="container">
+<div className='row'>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="Async api"
+schemaFormat: application/raml+yaml;version=1.0
+payload:
+ $ref: external-yaml.yaml#/definitions/User
+```
+
+</div>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="Yaml file (external-yaml.yaml)"
+definitions:
+ User:
+   type: string | object
+```
+
+</div>
+</div>
+</div>
+
+As previously mentioned, when the target file is a known raml fragment or api specification free use of relative json pointers is not supported.
+The following example shows an **invalid** example when using relative json pointer to reference an external RAML specification:
+
+<div className="container">
+<div className='row'>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="Async api"
+schemaFormat: application/raml+yaml;version=1.0
+payload:
+ $ref: ../api.raml#/types/User
+```
+
+</div>
+<div style={myStyle} className='col col--6'>
+
+```yaml title="External RAML api (api.raml)" {1}
+#%RAML 1.0
+title: Instagram API
+
+types:
+ User:
+   type: object
+```
+
+</div>
+</div>
+</div>
+
+
+
+## Inlined RAML types
+The simplest way to define a raml type in Async is by defining your content inlined under the `payload` facet while using the appropriate schema format like so:
+```yaml
+schemaFormat: application/raml+yaml;version=1.0
+payload: # The following is a RAML data type
+ type: object
+ properties:
+   a: object | string
+ additionalProperties: false
+```
+**All content within the payload will be considered strictly RAML**, except when using ‘$ref’ at the root of the payload (see the [previous section](#referencing-external-raml-content-using-ref) for more details).
+
+
+### Isolated RAML context
+When defining inlined types, its context is fully restricted from the content defined outside the payload facet.
+This means that any reference to content defined in the `components` facet of the main Async api will not be resolved.
+
+```yaml title="Referencing types defined in async api components (invalid)" {10}
+components:
+ schemas:
+   User: ...
+channels:
+ users/signup:
+   subscribe:
+     message:
+       schemaFormat: application/raml+yaml;version=1.0
+       payload:
+         type: User #unresolved reference
+```
+
+
+### Limited RAML referencing mechanism
+RAML has its own referencing mechanism to include content from external files using a reserved yaml tag `!include`.
+**AMF will only support referencing external data type fragments when using this mechanism.** This feature will be limited to Async apis defined in yaml format, as json has no way of defining tags.
```

Comment:
It's not immediately clear what "this feature" is referring to. I think the reader would likely know, but it may be good to specify.

Reply:
Thanks, it was definitely unclear. I've rewritten this paragraph and divided it into two different ones.